reactor-netty-http/src/main/java/reactor/netty/http/client/HttpClient.java

```diff
@@ -1634,6 +1635,35 @@ public Mono<Void> warmup() {
 				Mono.fromRunnable(OpenSsl::version));
 	}
 
+	/**
+	 * Supports proxy configuration with a deferred approach.
+	 *
+	 * <p>When proxyWhen(...) is set, calls to proxy(...) and noProxy() methods are ignored.</p>
+	 * <p>This method allows dynamic determination of proxy settings, applying or skipping the proxy based on the configured conditions.</p>
+	 *
+	 * @param proxyBuilder a consumer for proxy configuration
+	 * @return a new {@link HttpClient} reference
+	 */
+	public final HttpClient proxyWhen(BiConsumer<HttpClientConfig, ? super ProxyProvider.TypeSpec> proxyBuilder) {
+		Objects.requireNonNull(proxyBuilder, "proxyBuilder");
+		HttpClient dup = duplicate();
+		dup.configuration()
+				.deferredConf(
+						config -> {
+							ProxyProvider.TypeSpec spec = ProxyProvider.builder();
+							proxyBuilder.accept(config, spec);
+
+							if (((ProxyProvider.Validator) spec).isConfiguredCorrectly()) {
+								ProxyProvider proxyProvider = ((ProxyProvider.Builder) spec).build();
+								config.proxyProvider(proxyProvider);
+							}
+
+							return Mono.just(config);
+						}
+				);
+		return dup;
+	}
```

Comment:
What do you think if we use `Mono` for the proxy configuration? Imagine that you need to resolve something else before providing the proxy configuration.
Something similar to what we have for the headers e.g. this test case:
https://github.com/reactor/reactor-netty/blob/c5e4a96a297a2577fbc1a55f8255ac43f205b002/reactor-netty-http/src/test/java/reactor/netty/http/client/WebsocketClientOperationsTest.java#L85

```suggestion
	public final HttpClient proxyWhen(BiFunction<HttpClientConfig, ? super ProxyProvider.TypeSpec,
			Mono<? extends ProxyProvider.Builder>> proxyBuilder) {
		Objects.requireNonNull(proxyBuilder, "proxyBuilder");
		HttpClient dup = duplicate();
		dup.configuration().deferredConf(config -> {
			Mono<? extends ProxyProvider.Builder> mono = proxyBuilder.apply(config, ProxyProvider.builder());
			return mono == Mono.<ProxyProvider.Builder>empty() ?
					Mono.just(config) :
					mono.map(builder -> {
						config.proxyProvider(builder.build());
						return config;
					});
		});
		return dup;
	}
```

Reply:
@violetagg 
Great point!
I’ll modify it to use Mono as suggested, ensuring it operates without blocking.
Thank you for your valuable feedback! 😊

Reply:
@violetagg 
Hello, I’ve updated the code to reflect your feedback!
Please review the changes, and let me know if there are any additional improvements needed.
Thank you!